alibi/explainers/integrated_gradients.py

```diff
@@ -578,25 +852,30 @@ def _compute_attributions_list_input(self,
             Interpolation parameter defining the points of the interal path.
         nb_samples
             Total number of samples.
+        compute_layer_inputs_gradients
+            In case of layers gradients, controls whether the gradients are computed for the layer's inputs or
+            outputs. If True, gradients are computed for the layer's inputs, if False for the layer's outputs.
+
         Returns
         -------
-            Tuple with integrated gradients attribution and formatted baselines
+            Tuple with integrated gradients attributions, deltas and predictions
 
         """
-        # fix orginal call method for layer
-        if self.layer is not None:
-            orig_calls = self.layer.call
-        else:
-            orig_calls = None
+        input_dtypes = [xx.dtype for xx in X]
+
+        if (len(self.model.output_shape) == 1
+            or self.model.output_shape[-1] == 1) \
+                and target is None:
+            logger.warning("It looks like you are passing a model with a scalar output and target is set to `None`."
+                           "If your model is a regression model this will produce correct attributions. If your model "
+                           "is a classification model, targets for each datapoint must be defined. "
+                           "Not defining the target may lead to incorrect values for the attributions."
+                           "Targets can be either the true classes or the classes predicted by the model.")
```

Comment:
Can this check be factored out and performed earlier (ditto for the other tensor attribution function)? Two reasons:
- it's repeated in both `tensor` and `list` attribution functions
- we can do the validation before computing attributions as the model is called and `output_shape` is know, so this check should not be inside the attribution calculation code

Reply:
If the model doesn't have `inputs`, you have to call the model on a dummy input to the output shape. So the warning has to be repeated twice anyway in explain for list inputs and tensor inputs. I think it's cleaner to leave it inside the functions.

Reply:
Just leaving a note that this has been refactored out as part of `_validate_output`.